docs/vocabulary.md

```diff
@@ -0,0 +1,312 @@
+process. Could be given by reviewers or CI/CD processes.
+- **Submit**: An action that allows Gerrit to merge a Change to the Git
+repository.
+
+### GitHub
+
+[GitHub](https://www.github.com) is a hosted software development version
+control system using Git.
+
+Some of the core GitHub terms are listed below. [[11]]
+
+- **Pull Request**: Proposed changes to a repository submitted by a user
+and accepted or rejected by a repository's collaborators.
+- **Review**: Process that enables peer-review to improve the quality of
+code.
+- **Status Check**: Status checks are external processes, such as continuous
+integration builds, which run for each commit you make in a repository.
+- **Check**: A check is a type of status check on GitHub.
+- **Merge**: Acceptance of a PR by the maintainers.
+
+### GitLab
+
+[GitLab](https://www.gitlab.com) is a web-based DevOps lifecycle tool that
+provides a Git-repository manager providing wiki, issue-tracking and CI/CD
+pipeline features.
+
+Some of the core GitLab terms are listed below. [[12]]
+
+- **Merge Request**: MR is the basis of GitLab as a code collaboration and
+version control platform. It is as simple as the name implies: a request to
+merge one branch into another.
+- **Review**: Process that enables peer-review to improve the quality of
+code.
+- **Merge Check**: Checks to ensure MR can be merged without issues.
+- **Merge**: Acceptance of a MR by the maintainers.
+
+## Mapping of Terms
+
+The CI/CD Tools and Technologies listed in this document use more or less
+same terminology for naming things with few differences, workflow vs pipeline
+or runner vs node except the smaller units of work.
+
+The table below is an attempt to create a mapping of different terms used
+by CI/CD Tools and Technologies and could contain errors. The main term used
+while creating the table is **pipeline** since almost all the tools listed in
+this document use the word pipeline to describe pretty similar things. Smaller
+units of terms then mapped based on how they are described in relation to pipeline
+in corresponding documentation.
+
+| Project            |        |         |         |          |         |              |
+|--------------------|--------|---------|---------|----------|---------|--------------|
+| **CircleCI**       | N/A    | Step    | Job     | Workflow | Trigger | Executor     |
+| **GitHub Actions** | Action | Step    | Job     | Workflow | Event   | Runner       |
+| **GitLab CI/CD**   | N/A    | Job     | Stage   | Pipeline | Trigger | Runner       |
+| **Jenkins**        | N/A    | Project | Stage   | Pipeline | Trigger | Node         |
+| **Jenkins X**      | N/A    | Step    | Stage   | Pipeline | Trigger | Agent        |
+| **Screwdriver**    | N/A    | Step    | Job     | Pipeline | Trigger | N/A          |
+| **Spinnaker**      | N/A    | Task    | Stage   | Pipeline | Trigger | Cluster      |
+| **Tekton**         | N/A    | Step    | Task    | Pipeline | Trigger | Resource (?) |
```

Comment:
Resource isn't the right value for the last column - maybe "node"? I'm not 100% sure what the last column is meant to represent tho

Might help to choose a term for the top of each column and define it so it's clear what each column is?

(e.g. i'm not exactly sure what to map to "action" since I'm not sure what that column is either)

Reply:
> Resource isn't the right value for the last column - maybe "node"? I'm not 100% sure what the last column is meant to represent tho

The trouble I had about the "resource" was what "resource" means for Tekton and Spinnakerso I just included them here with the hope that we can either come up with a new term for the resources that are not same as "node" in Jenkins/GitLab and or add additional text explaining our reasoning. We can fix this in a separate PR.
> 
> Might help to choose a term for the top of each column and define it so it's clear what each column is?
> 
> (e.g. i'm not exactly sure what to map to "action" since I'm not sure what that column is either)

This is similar to "resource" column - I wasn't sure how to name columns and left them empty so we can determine the name for them.